ambry-cloud/src/main/java/com.github.ambry.cloud/CloudDataNode.java

```diff
@@ -51,7 +51,7 @@
    * @param cloudConfig the {@link CloudConfig} to use.
    * @param clusterMapConfig the {@link ClusterMapConfig} to use.
    */
-  CloudDataNode(CloudConfig cloudConfig, ClusterMapConfig clusterMapConfig) {
+  public CloudDataNode(CloudConfig cloudConfig, ClusterMapConfig clusterMapConfig) {
     this.hostName = clusterMapConfig.clusterMapHostName;
     this.plainTextPort = new Port(clusterMapConfig.clusterMapPort, PortType.PLAINTEXT);
```

Comment:
This can throw NPE if clusterMapPort is null.  Would be better to throw IllegalArgumentException with clear message.  Same for vcrSslPort on next line.